### ISSUE TITLE
Home assistant disconnecting. Memory Leak?

### ISSUE THREAD

**Issue by mendoncart** [node-red/node-red#4543]
### Current Behavior

From time to time, all the flows stop working.
Events don't trigger the flows.
And to top it all off, Home Assistant is disconnecting...

It all started when I upgraded the container version.
Current version: 3.1.3-18

Checking the container logs, I've this:
```
<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
./entrypoint.sh: line 14:     8 Aborted                 (core dumped) /usr/local/bin/node $NODE_OPTIONS node_modules/node-red/red.js --userDir /data $FLOWS "${@}"
24 Jan 11:23:53 - [info] 
```

Full Logs
```
24 Jan 11:17:51 - [info] [server:Home Assistant] Connecting to http://192.168.15.202:8123
24 Jan 11:17:55 - [info] [server:Home Assistant] Connected to http://192.168.15.202:8123
24 Jan 11:17:56 - [info] [server:Home Assistant] Connection closed to http://192.168.15.202:8123
24 Jan 11:17:58 - [info] [server:Home Assistant] Connecting to http://192.168.15.202:8123
24 Jan 11:18:16 - [info] [server:Home Assistant] Connected to http://192.168.15.202:8123
24 Jan 11:18:18 - [info] [server:Home Assistant] Connection closed to http://192.168.15.202:8123
24 Jan 11:18:20 - [info] [server:Home Assistant] Connecting to http://192.168.15.202:8123
24 Jan 11:18:23 - [info] Stopping flows
24 Jan 11:18:23 - [info] [server:Home Assistant] Closing connection to http://192.168.15.202:8123
24 Jan 11:18:28 - [info] Stopped flows
24 Jan 11:18:28 - [info] Updated flows
24 Jan 11:18:33 - [info] Starting flows
24 Jan 11:18:34 - [info] Started flows
24 Jan 11:18:34 - [info] [server:Home Assistant] Connecting to http://192.168.15.202:8123
24 Jan 11:18:39 - [info] [server:Home Assistant] Connected to http://192.168.15.202:8123
24 Jan 11:18:58 - [error] [trigger-state:Anoitecendo] ConfigError: Entity "Obrigatóio" not found
24 Jan 11:19:52 - [info] [server:Home Assistant] Connection closed to http://192.168.15.202:8123
24 Jan 11:19:52 - [error] [api-call-service:Desliga Luz] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
24 Jan 11:19:53 - [info] [server:Home Assistant] Connecting to http://192.168.15.202:8123
24 Jan 11:19:53 - [info] [server:Home Assistant] Connected to http://192.168.15.202:8123
24 Jan 11:20:20 - [info] [server:Home Assistant] Connection closed to http://192.168.15.202:8123
24 Jan 11:20:29 - [info] [server:Home Assistant] Connecting to http://192.168.15.202:8123
24 Jan 11:20:30 - [info] [server:Home Assistant] Connected to http://192.168.15.202:8123
24 Jan 11:20:34 - [info] [server:Home Assistant] Connection closed to http://192.168.15.202:8123
24 Jan 11:20:34 - [info] [server:Home Assistant] Connecting to http://192.168.15.202:8123
24 Jan 11:20:34 - [info] [server:Home Assistant] Connected to http://192.168.15.202:8123
24 Jan 11:20:59 - [info] [server:Home Assistant] Connection closed to http://192.168.15.202:8123
24 Jan 11:21:13 - [info] [server:Home Assistant] Connecting to http://192.168.15.202:8123
24 Jan 11:22:21 - [info] [server:Home Assistant] Connecting to http://192.168.15.202:8123

<--- Last few GCs --->

[8:0x7faa9c760030] 86904031 ms: Scavenge 3974.1 (4124.1) -> 3965.9 (4127.6) MB, 69.3 / 0.0 ms  (average mu = 0.203, current mu = 0.034) allocation failure; 
[8:0x7faa9c760030] 86904358 ms: Scavenge 3977.3 (4127.6) -> 3969.4 (4130.6) MB, 92.4 / 0.0 ms  (average mu = 0.203, current mu = 0.034) allocation failure; 
[8:0x7faa9c760030] 86914165 ms: Mark-sweep 3981.1 (4130.6) -> 3963.3 (4134.6) MB, 9589.3 / 0.0 ms  (average mu = 0.166, current mu = 0.131) allocation failure; scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
./entrypoint.sh: line 14:     8 Aborted                 (core dumped) /usr/local/bin/node $NODE_OPTIONS node_modules/node-red/red.js --userDir /data $FLOWS "${@}"
24 Jan 11:23:53 - [info] 

Welcome to Node-RED
===================

24 Jan 11:23:53 - [info] Node-RED version: v3.1.3
24 Jan 11:23:53 - [info] Node.js  version: v18.19.0
24 Jan 11:23:53 - [info] Linux 4.19.0-26-amd64 x64 LE
24 Jan 11:23:54 - [info] Loading palette nodes
24 Jan 11:24:26 - [info] Settings file  : /data/settings.js
24 Jan 11:24:26 - [info] Context store  : 'default' [module=memory]
24 Jan 11:24:26 - [info] User directory : /data
24 Jan 11:24:26 - [warn] Projects disabled : editorTheme.projects.enabled=false
24 Jan 11:24:26 - [info] Flows file     : /data/flows.json
24 Jan 11:24:26 - [info] Server now running at http://127.0.0.1:1880/
24 Jan 11:24:26 - [warn] 

---------------------------------------------------------------------
Your flow credentials file is encrypted using a system-generated key.

If the system-generated key is lost for any reason, your credentials
file will not be recoverable, you will have to delete it and re-enter
your credentials.

You should set your own key using the 'credentialSecret' option in
your settings file. Node-RED will then re-encrypt your credentials
file using your chosen key the next time you deploy a change.
---------------------------------------------------------------------

24 Jan 11:24:26 - [info] Starting flows
24 Jan 11:24:27 - [info] Started flows
24 Jan 11:24:27 - [info] [server:Home Assistant] Connecting to http://192.168.15.202:8123
24 Jan 11:24:27 - [info] [server:Home Assistant] Connected to http://192.168.15.202:8123
24 Jan 11:24:28 - [error] [trigger-state:Anoitecendo] ConfigError: Entity "Obrigatóio" not found

```

### Expected Behavior

To everything keep working like it was.

### Steps To Reproduce

Update the container version to 3.1.3-18.

### Example flow

All flows I already had.

### Environment

- Node-RED version: 3.1.3-18
- Home Assistant: 2023.12.3
- Platform/OS: Docker
- Browser: Brave

**Comment by knolleary** [comment]
What container version were you using previously?

**Comment by hardillb** [comment]
`latest` will have pointed to the `3.0.x` releases until the `3.1.0` was released, so it will depend on when you last upgraded.

the `latest` branch is also NodeJS 16 based (this will change at NR 4.0) so changing to `latest-18` will have moved over from NodeJS 16 to NodeJS 18.

If the image is present, then you should be able to inspect it and it will list it's version in the labels.

**Comment by mendoncart** [comment]
It has been some time since I installed node-red, so it's a relatively old image.

Here's the portainer information about the image:
![image](https://github.com/node-red/node-red/assets/62950083/251d5064-9b3e-468b-a92e-b10c35bd807d)


<html>
<body>


<table border="1" cellpadding="1" cellspacing="1" style="width:765px">
	<tbody>
		<tr>
			<td>Order</td>
			<td>&nbsp;Size</td>
			<td style="width:696px">
			<p>Layer</p>
			</td>
		</tr>
		<tr>
			<td>1</td>
			<td>7 MB</td>
			<td style="width:696px">
			<p>ADD file:9a4f77dfaba7fd2aa78186e4ef0e7486ad55101cefc1fabbc1b385601bb38920 in /</p>
			</td>
		</tr>
		<tr>
			<td>2</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>CMD [&quot;/bin/sh&quot;]</p>
			</td>
		</tr>
		<tr>
			<td>3</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ENV NODE_VERSION=16.20.0</p>
			</td>
		</tr>
		<tr>
			<td>4</td>
			<td>102.4 MB</td>
			<td style="width:696px">
			<p>RUN addgroup -g 1000 node &amp;&amp; adduser -u 1000 -G node -s /bin/sh -D node &amp;&amp; apk add --no-cache libstdc++ &amp;&amp; ...&nbsp;</p>
			</td>
		</tr>
		<tr>
			<td>5</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ENV YARN_VERSION=1.22.19</p>
			</td>
		</tr>
		<tr>
			<td>6</td>
			<td>7.8 MB</td>
			<td style="width:696px">
			<p>RUN apk add --no-cache --virtual .build-deps-yarn curl gnupg tar &amp;&amp; for key in 6A010C5166006599AA17F08146C2130DFD2497F5 ...&nbsp;</p>
			</td>
		</tr>
		<tr>
			<td>7</td>
			<td>388 B</td>
			<td style="width:696px">
			<p>COPY file:4d192565a7220e135cab6c77fbc1c73211b69f3d9fb37e62857b2c6eb9363d51 in /usr/local/bin/</p>
			</td>
		</tr>
		<tr>
			<td>8</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ENTRYPOINT [&quot;docker-entrypoint.sh&quot;]</p>
			</td>
		</tr>
		<tr>
			<td>9</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>CMD [&quot;node&quot;]</p>
			</td>
		</tr>
		<tr>
			<td>10</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ARG QEMU_ARCH</p>
			</td>
		</tr>
		<tr>
			<td>11</td>
			<td>751 B</td>
			<td style="width:696px">
			<p>COPY .docker/scripts/*.sh /tmp/ # buildkit</p>
			</td>
		</tr>
		<tr>
			<td>12</td>
			<td>878 B</td>
			<td style="width:696px">
			<p>COPY .docker/healthcheck.js / # buildkit</p>
			</td>
		</tr>
		<tr>
			<td>13</td>
			<td>18.2 MB</td>
			<td style="width:696px">
			<p>RUN |1 QEMU_ARCH= RUN set -ex &amp;&amp; apk add --no-cache bash tzdata iputils curl nano ...&nbsp;</p>
			</td>
		</tr>
		<tr>
			<td>14</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>WORKDIR /usr/src/node-red</p>
			</td>
		</tr>
		<tr>
			<td>15</td>
			<td>2.9 kB</td>
			<td style="width:696px">
			<p>COPY .docker/known_hosts.sh . # buildkit</p>
			</td>
		</tr>
		<tr>
			<td>16</td>
			<td>3.1 kB</td>
			<td style="width:696px">
			<p>RUN |1 QEMU_ARCH= RUN ./known_hosts.sh /etc/ssh/ssh_known_hosts &amp;&amp; rm /usr/src/node-red/known_hosts.sh # buildkit</p>
			</td>
		</tr>
		<tr>
			<td>17</td>
			<td>1.6 kB</td>
			<td style="width:696px">
			<p>RUN |1 QEMU_ARCH= RUN echo &quot;PubkeyAcceptedKeyTypes +ssh-rsa&quot; &gt;&gt; /etc/ssh/ssh_config # buildkit</p>
			</td>
		</tr>
		<tr>
			<td>18</td>
			<td>1 kB</td>
			<td style="width:696px">
			<p>COPY package.json . # buildkit</p>
			</td>
		</tr>
		<tr>
			<td>19</td>
			<td>1.4 kB</td>
			<td style="width:696px">
			<p>COPY flows.json /data # buildkit</p>
			</td>
		</tr>
		<tr>
			<td>20</td>
			<td>226 B</td>
			<td style="width:696px">
			<p>COPY .docker/scripts/entrypoint.sh . # buildkit</p>
			</td>
		</tr>
		<tr>
			<td>21</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ARG BUILD_DATE</p>
			</td>
		</tr>
		<tr>
			<td>22</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ARG BUILD_VERSION</p>
			</td>
		</tr>
		<tr>
			<td>23</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ARG BUILD_REF</p>
			</td>
		</tr>
		<tr>
			<td>24</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ARG NODE_RED_VERSION</p>
			</td>
		</tr>
		<tr>
			<td>25</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ARG ARCH</p>
			</td>
		</tr>
		<tr>
			<td>26</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ARG QEMU_ARCH</p>
			</td>
		</tr>
		<tr>
			<td>27</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ARG TAG_SUFFIX</p>
			</td>
		</tr>
		<tr>
			<td>28</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>LABEL org.label-schema.build-date=2023-05-30T09:15:08Z org.label-schema.docker.dockerfile=.docker/Dockerfile.alpine org.label-s...&nbsp;</p>
			</td>
		</tr>
		<tr>
			<td>29</td>
			<td>55.2 MB</td>
			<td style="width:696px">
			<p>COPY /usr/src/node-red/prod_node_modules ./node_modules # buildkit</p>
			</td>
		</tr>
		<tr>
			<td>30</td>
			<td>350.5 MB</td>
			<td style="width:696px">
			<p>RUN |7 BUILD_DATE=2023-05-30T09:15:08Z BUILD_VERSION=3.0.2 BUILD_REF= NODE_RED_VERSION=v3.0.2 ARCH= QEMU_ARCH= TAG_SUFFIX= RUN ...&nbsp;</p>
			</td>
		</tr>
		<tr>
			<td>31</td>
			<td>1.8 kB</td>
			<td style="width:696px">
			<p>RUN |7 BUILD_DATE=2023-05-30T09:15:08Z BUILD_VERSION=3.0.2 BUILD_REF= NODE_RED_VERSION=v3.0.2 ARCH= QEMU_ARCH= TAG_SUFFIX= RUN ...&nbsp;</p>
			</td>
		</tr>
		<tr>
			<td>32</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>USER node-red</p>
			</td>
		</tr>
		<tr>
			<td>33</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ENV NODE_RED_VERSION=v3.0.2 NODE_PATH=/usr/src/node-red/node_modules:/data/node_modules PATH=/usr/src/node-red/node_modules/.bi...&nbsp;</p>
			</td>
		</tr>
		<tr>
			<td>34</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>EXPOSE map[1880/tcp:{}]</p>
			</td>
		</tr>
		<tr>
			<td>35</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>HEALTHCHECK &amp;{[&quot;CMD-SHELL&quot; &quot;node /healthcheck.js&quot;] &quot;0s&quot; &quot;0s&quot; &quot;0s&quot; &#39;\x00&#39;}</p>
			</td>
		</tr>
		<tr>
			<td>36</td>
			<td>0 B</td>
			<td style="width:696px">
			<p>ENTRYPOINT [&quot;./entrypoint.sh&quot;]</p>
			</td>
		</tr>
	</tbody>
</table>

<p>&nbsp;</p>



</body>
</html>

**Comment by hardillb** [comment]
OK, so if you've moved from NodeJS 16 to 18 you will need to run `npm rebuild` in the `/data` directory to make sure all the nodes with any native components are updated. You'll need to open a terminal in the container to do this.

After that then it does look like a memory leak, so you'll have to try and track that down to either your flow or any of the nodes you have installed.

**Comment by mendoncart** [comment]
I didn't change anything but the version upgrade.
All the flows are the same. I upgraded the home assistant node-red contribs pallet and updated the flows.
Isn't this memory leak something wrong in the version I've installed?

**Comment by hardillb** [comment]
Ensure the rebuild is done first. That needs to be done.

Then you've said you changed 2 things:

 - The Node-RED version 
 - The home assistant nodes version

We need to separate these to work out where the problem is, but so far we've not had any reports of a memory leak in Node-RED, so I worked start with the nodes

**Comment by mendoncart** [comment]
I updated to the latest tag and the memory leak seems to have disappeared. 
I'll come back later and update this issue accordingly.

Now I'll search why the connection with Home Assistant is being closed and re-established repeatedly.

**Comment by hardillb** [comment]
The current `latest` tag is Node-RED 3.1.3 with NodeJS 16

**Comment by mendoncart** [comment]
> The current latest tag is Node-RED 3.1.3 with NodeJS 16

I'll stay on this version for now, thanks!
If everything keeps working smoothly, I'll close this issue.
HA disconnecting seems to be related to the pallet, as seen on issue [1225](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/issues/1225) on the pallet page.

**Comment by knolleary** [comment]
I'm going to close this issue off - let us know if you are still seeing any issues.

**Comment by mendoncart** [comment]
Oh, sorry I forgot to give feedback.
I've rolled back the version and the memory leak stopped, the only thing is Home Assistant keeps disconnecting, but this is a palette issue, not a node-red one.

Thanks to all who helped me!

**Comment by Nornode** [comment]
Hey... I'm sorry to comment on an old issue.
I seem to have the same trouble and can't figure out what is causing this.
I managed to keep the container running by adding additional memory to it.

This is the logs I get. 

```
Welcome to Node-RED
===================

3 Dec 22:44:07 - [info] Node-RED version: v4.0.5
3 Dec 22:44:07 - [info] Node.js  version: v20.18.0
3 Dec 22:44:07 - [info] Linux 5.15.0-125-generic x64 LE
3 Dec 22:44:08 - [info] Loading palette nodes
3 Dec 22:44:09 - [info] node-red-contrib-home-assistant-websocket v0.74.2 nodes initialized
3 Dec 22:44:09 - [info] Settings file  : /data/settings.js
3 Dec 22:44:09 - [info] Context store  : 'default' [module=memory]
3 Dec 22:44:09 - [info] User directory : /data
3 Dec 22:44:09 - [warn] Projects disabled : editorTheme.projects.enabled=false
3 Dec 22:44:09 - [info] Flows file     : /data/flows.json
3 Dec 22:44:09 - [info] Server now running at http://127.0.0.1:1880/
3 Dec 22:44:09 - [warn]

---------------------------------------------------------------------
Your flow credentials file is encrypted using a system-generated key.
...
---------------------------------------------------------------------

3 Dec 22:44:10 - [info] Starting flows
3 Dec 22:44:10 - [info] Started flows
3 Dec 22:44:10 - [info] [server:FQDN.of.server - Local IP] Connecting to http://192.168.10.15:8123
3 Dec 22:44:10 - [info] [server:FQDN.of.server - Local IP] Connected to http://192.168.10.15:8123
3 Dec 22:44:13 - [info] [ps-strategy-lowest-price:Lowest Price Floors 4h] Switching on in 8146513 milliseconds
3 Dec 22:44:13 - [info] [ps-strategy-lowest-price:Lowest Price Boiler 3h] Switching on in 8146490 milliseconds
3 Dec 22:44:13 - [info] [ps-strategy-lowest-price:Lowest Price Ventilation 6h] Switching on in 4546467 milliseconds
3 Dec 22:44:13 - [info] [ps-strategy-lowest-price:ON for 4 cheapest] Switching on in 4546434 milliseconds
3 Dec 22:44:15 - [info] [ps-strategy-lowest-price:3h - All LOW and OFF @ High hours - Normal Setting 21h on] Switching off in 33344691 milliseconds
3 Dec 22:44:15 - [info] [ps-strategy-lowest-price:Cheapest 3h (max ¢25) -> Higher target Temp - Rest Lower] Switching on in 8144678 milliseconds
3 Dec 22:44:15 - [info] [ps-strategy-lowest-price:Lowest € - Basement - 8h] Switching on in 944667 milliseconds
3 Dec 22:45:57 - [info] [ps-strategy-lowest-price:Lowest Price Floors 4h] Switching on in 8042896 milliseconds
3 Dec 22:45:57 - [info] [ps-strategy-lowest-price:3h - All LOW and OFF @ High hours - Normal Setting 21h on] Switching off in 33242441 milliseconds
3 Dec 22:45:57 - [info] [ps-strategy-lowest-price:Cheapest 3h (max ¢25) -> Higher target Temp - Rest Lower] Switching on in 8042428 milliseconds
3 Dec 22:45:57 - [info] [ps-strategy-lowest-price:Lowest € - Basement - 8h] Switching on in 842417 milliseconds
3 Dec 22:46:15 - [info] [server:FQDN.of.server - Local IP] Connection closed to http://192.168.10.15:8123
3 Dec 22:46:15 - [error] [api-call-service:Library Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:MBR Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Main Entrance Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Kitchen Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Livingroom Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Castalia Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Clarissa Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Zacke Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:UpperBath Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Aula Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Library Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:MBR Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Main Entrance Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Kitchen Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Livingroom Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Castalia Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Clarissa Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:Zacke Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [error] [api-call-service:UpperBath Low Setting] Error: Unrecognized error: {"type":"result","success":false,"error":{"code":3,"message":"Connection lost"}}
3 Dec 22:46:15 - [info] [server:FQDN.of.server - Local IP] Connecting to http://192.168.10.15:8123
3 Dec 22:46:15 - [info] [server:FQDN.of.server - Local IP] Connected to http://192.168.10.15:8123
```

Also the container specs:
```
version=4.0.5
docker run --init -d \
  --name=$name \
  --net=bridge \
  -p 1880:1880 \
  -e NODE_OPTIONS="--max_old_space_size=512" \
  -v /docker/$name:/data \
  -v /etc/timezone:/etc/timezone:ro \
  -e TZ=Europe/Helsinki \
  --restart unless-stopped \
  --log-opt max-size=1m \
  nodered/node-red:$version

docker logs -f $name
```

Same thing seemed to happen with the HA add-on which was why I moved back to an independent container connecting back to HA (on 192.168.10.15).

Does the amount of devices/entities in HA play a role here?
